java/avro/src/main/java/com/microsoft/azure/schemaregistry/kafka/avro/KafkaAvroDeserializer.java

```diff
@@ -48,12 +50,23 @@ public KafkaAvroDeserializer() {
      */
     public void configure(Map<String, ?> props, boolean isKey) {
         this.config = new KafkaAvroDeserializerConfig((Map<String, Object>) props);
+        TokenCredential tokenCredential;
+        tokenCredential = this.config.getCredential();
+        if (tokenCredential == null && this.config.createDefaultAzureCredential()) {
+            tokenCredential = new DefaultAzureCredentialBuilder().build();
+        } else {
```

Comment:
won't this means that if tokenCredential is not null to begin with we end up throwing RuntimeException?